tic_tac_toe.py

```diff
@@ -0,0 +1,111 @@
+import random
+
+
+class TicTacToe:
+
+    def __init__(self):
+        self.board = [None]*9
+
+    def make_move(self, row: int, col: int, player):
+        position = col*3 + row
+        self.board[position] = player
+        return
+
+    def make_random_move(self, player):
+        available_squares = []
+        for i in range(9):
+            if self.board[i] is None:
+                available_squares.append(i)
+        position = random.choice(available_squares)
+        self.board[position] = player
+        return
+
+    def _get_position(self, row, col):
+        return col*3 + row
+
+    def _get_board_state(self, row, col):
+        position = self._get_position(row, col)
+        return self.board[position]
+
+    def _check_diagonal_win(self):
+        center_square = self._get_board_state(1, 1)
+        if not center_square:
+            return None
+
+        if ((center_square == self._get_board_state(0, 0)) and
+                (center_square == self._get_board_state(2, 2))):
+            return center_square
+        elif ((center_square == self._get_board_state(0, 2)) and
+              (center_square == self._get_board_state(2, 0))):
+            return center_square
+
+        return None
+
+    def _check_row_win(self):
+        for row in (0, 1, 2):
+            if (self._get_board_state(row, 0) ==
+                self._get_board_state(row, 1) ==
+                    self._get_board_state(row, 2)):
+                return self._get_board_state(row, 0)
+
+        return None
+
+    def _check_col_win(self):
+        for col in (0, 1, 2):
+            if (self._get_board_state(0, col) ==
+                self._get_board_state(1, col) ==
+                    self._get_board_state(2, col)):
+                return self._get_board_state(0, col)
+
+        return None
+
+    def check_win(self):
+        for check_winner in (self._check_diagonal_win,
+                             self._check_row_win,
+                             self._check_col_win):
+            winner = check_winner()
+            if winner:
+                return winner
+
+        return None
+
+    def display_board(self):
+        board = [square or ' ' for square in self.board]
+
+        template = f"""
+          {board[0]} | {board[1]} | {board[2]}
+         --- --- ---
+          {board[3]} | {board[4]} | {board[5]}
+         --- --- ---
+          {board[6]} | {board[7]} | {board[8]}
+        """
+
+        print(template)
+
+        return
+
+
+def main():
+    ttt = TicTacToe()
+
+    while True:
+        ttt.display_board()
+        print("Player X, choose your next move!")
```

Comment:
Instruction for user  is not clear enough.  Would be better to show user the format of input string, and show the range of each param (also changing valid value of row and col from `0-2` to `1-3` could make it more human-friendly)

For example, the instruction message can be:
```
Player X, choose your next move!
Please choose a cell with row 0-2 and column 0-2.
For example, enter the following input for making a move on the upper left corner (first row first column):
0,0
```